services/github/github-actions.service.js

```diff
@@ -0,0 +1,107 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { renderBuildStatusBadge } = require('../build-status')
+const { nonNegativeInteger } = require('../validators')
+const { GithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+const { NotFound } = require('..')
+
+const schema = Joi.object({
+  total_count: nonNegativeInteger,
+  check_suites: Joi.array()
+    .items(
+      Joi.object({
+        status: Joi.string().required(),
+        conclusion: Joi.string()
+          .required()
+          .allow(null),
+        app: Joi.object({
+          id: nonNegativeInteger,
+          name: Joi.string().required(),
+        }).required(),
+      })
+    )
+    .min(0)
+    .required(),
+}).required()
+
+module.exports = class GitHubActions extends GithubAuthV3Service {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'github/actions',
+      pattern: ':user/:repo/:ref',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub Actions',
+        namedParams: {
+          user: 'actions',
+          repo: 'setup-node',
+          ref: 'master',
+        },
+        staticPreview: renderBuildStatusBadge({ status: 'success' }),
+        documentation: `
+          You can use a branch name, tag, or commit SHA for the ref parameter.
+          ${documentation}
+        `,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'workflow',
+    }
+  }
+
+  static render({ status, conclusion }) {
+    if (status !== 'completed') {
+      return {
+        message: 'in progress',
```

Comment:
`in progress` should probably be added to the list of `otherStatuses` in _build-status.js_. The test 'valid repo and ref' may occasionally fail otherwise. I would also suggest using `renderBuildStatusBadge` for the 'in progress' value as well: it won't change anything with the current implementation, but if ever we decide to change the colour or something else, this build badge will automatically be covered as well. 😉